File: 251003/Stasevich/Publisher/Controller/UserController.cs
using Microsoft.AspNetCore.Mvc;
using WebApplication1.DTO;
using WebApplication1.Repository;
using WebApplication1.Service;

namespace WebApplication1.Controller
{
    [ApiController]
    [Route("api/v1.0/users")]
    public class UserController : ControllerBase
    {
        private readonly IUserService _userService;
        public UserController(IUserService userService) => _userService = userService;

        [HttpGet]
        public async Task<ActionResult<IEnumerable<UserResponseTo>>> GetAll()
        {
            var allUsers = await _userService.GetAllUsersAsync(1, 1000);
            return Ok(allUsers.Items);
        }

        [HttpGet("{id:long}")]
        public async Task<ActionResult<UserResponseTo>> GetById(long id)
        {
            var user = await _userService.GetUserByIdAsync(id);
            return Ok(user);
        }

        [HttpPost]
        public async Task<ActionResult<UserResponseTo>> Create([FromBody] UserRequestTo dto)
        {
            var created = await _userService.CreateUserAsync(dto);
            return CreatedAtAction(nameof(GetById), new { id = created.id }, created);
        }

        [HttpPut]
        public async Task<ActionResult<UserResponseTo>> Update([FromBody] UserRequestTo dto)
        {
            if (dto.Id == null)
                return BadRequest("Id must be provided in the request body");
            var updated = await _userService.UpdateUserAsync(dto.Id.Value, dto);
            return Ok(updated);
        }

        [HttpDelete("{id:long}")]
        public async Task<IActionResult> Delete(long id)
        {
            try
            {
                await _userService.DeleteUserAsync(id);
                return NoContent();
            }
            catch (ValidationException ex)
            {
                return StatusCode(ex.HttpCode, new { error = ex.Message, code = ex.ErrorCode });
            }
        }
    }
}

File: 251001/Brutskaya/disCo/publisher/src/main/java/com/example/publisher/api/controller/UserController.java
package com.example.publisher.api.controller;

import com.example.publisher.api.dto.request.UserRequestTo;
import com.example.publisher.api.dto.responce.UserResponseTo;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.ResponseStatus;

import java.util.List;

public interface UserController {

    @PostMapping("/api/v1.0/users")
    @ResponseStatus(HttpStatus.CREATED)
    UserResponseTo create(@RequestBody
                          @Valid UserRequestTo request);

    @GetMapping("/api/v1.0/users")
    @ResponseStatus(HttpStatus.OK)
    List<UserResponseTo> getAll();

    @GetMapping("/api/v1.0/users/{id}")
    @ResponseStatus(HttpStatus.OK)
    UserResponseTo getById(@PathVariable Long id);

    @PutMapping("/api/v1.0/users")
    @ResponseStatus(HttpStatus.OK)
    UserResponseTo update(@RequestBody
                          @Valid UserRequestTo request);

    @DeleteMapping("/api/v1.0/users/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    void delete(@PathVariable Long id);
}
